packages/perseus/src/widgets/expression/validate-expression.ts

```diff
@@ -0,0 +1,28 @@
+import type {PerseusStrings} from "../../strings";
+import type {PerseusScore} from "../../types";
+import type {
+    PerseusExpressionValidationData,
+    PerseusExpressionUserInput,
+} from "../../validation.types";
+
+/**
+ * Checks user input from the expression widget to see if it is gradable. Most
+ * of the expression widget's validation requires the Rubric because of its
+ * use of KhanAnswerTypes as a core part of scoring.
+ *
+ * @see `scoreExpression()` for more details.
+ */
+function validateExpression(
+    userInput: PerseusExpressionUserInput,
+    validationData: PerseusExpressionValidationData,
+    strings: PerseusStrings,
+    locale: string,
+): Extract<PerseusScore, {type: "invalid"}> | null {
+    if (userInput === "") {
+        return {type: "invalid", message: null};
+    }
+
+    return null;
+}
```

Comment:
Just my $0.02 CAD, my preference would be to make these consistent with what we did for the scorers - not include unused params in the definition:

``` TS
export type WidgetScorerFunction = (
    userInput: UserInput,
    rubric: Rubric,
    string?: PerseusStrings,
    locale?: string,
) => PerseusScore;

export type WidgetExports<
    T extends React.ComponentType<any> & Widget = React.ComponentType<any>,
> = Readonly<{
    // ...
    scorer?: WidgetScorerFunction;
    // ...
}>;

function scoreSorter(
    userInput: PerseusSorterUserInput,
    rubric: PerseusSorterRubric,
): PerseusScore {
    // ...
}

export default {
    name: "sorter",
    displayName: "Sorter",
    widget: Sorter,
    isLintable: true,
    scorer: scoreSorter,
} satisfies WidgetExports<typeof Sorter>;
```

I just like that it's less noise which makes it easier to find uses of things and to get a tl;dr of what the function is doing by looking at the signature.

```suggestion
function validateExpression(
    userInput: PerseusExpressionUserInput,
): Extract<PerseusScore, {type: "invalid"}> | null {
    if (userInput === "") {
        return {type: "invalid", message: null};
    }

    return null;
}
```

Not a blocker and not something we need to figure out now.